sympy/solvers/solvers.py

```diff
@@ -1557,7 +1557,11 @@ def _expand(p):
                         try:
                             t = Dummy('t')
                             iv = _solve(u - t, symbol, **flags)
-                            soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            new_soln = list(ordered({i.subs(t, s) for i in iv for s in soln}))
+                            if new_soln[0] == S.NaN:
```

Comment:
Actually, you don't even need `new_soln` - you can just use `soln` (like in the original line 1560), remove `nan`s from it and let `_solve` continue its thing (you'll probably need a flag because if there were no `nan`s, then we've already found all solutions). But that's a minor thing.